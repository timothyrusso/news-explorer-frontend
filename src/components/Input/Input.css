.popup__input {
  display: block;
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  max-width: 358px;
  width: 100%;
  margin: 0 auto 29.74px;
  padding: 0 0 9px;
  color: #000;
  opacity: 0.2;
}

@media screen and (max-width: 552px) {
  .popup__input {
    max-width: 238px;
    margin-bottom: 30px;
  }
}

.popup__input_type_error {
  color: rgba(255, 0, 0, 1);
  margin: 0;
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 1;
  width: 100%;
  max-width: 358px;
  display: block;
  position: relative;
  left: 36px;
  bottom: 25px;
}

@media screen and (max-width: 552px) {
  .popup__input_type_error {
    left: 22px;
  }
}

.popup__input-label {
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
  color: #2F71E5;
  margin: 0 0 9px 36px;
  display: block;
}

input:focus {
  color: #000;
  opacity: 1;
}
