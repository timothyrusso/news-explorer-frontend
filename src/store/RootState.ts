import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
