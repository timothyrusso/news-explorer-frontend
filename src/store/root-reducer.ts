import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
