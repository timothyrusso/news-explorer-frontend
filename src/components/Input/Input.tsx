import React, { FC } from 'react';
import { ErrorMessage } from '../../store/errors/error.type';
import './Input.css';

type InputProps = {
  type: string;
  idName: string;
  name: string;
  fieldName: string;
  placeholder: string;
  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  minLength?: number;
  maxLength?: number;
  value: string;
  errorMessage: ErrorMessage | {};
  labelText: string;
};

const Input: FC<InputProps> = ({
  type,
  idName,
  name,
  fieldName,
  placeholder,
  onChange,
  minLength,
  maxLength,
  value,
  errorMessage,
  labelText,
}) => {
  return (
    <>
      <label htmlFor={idName} className="popup__input-label">
        {labelText}
      </label>
      <input
        type={type}
        id={idName}
        name={name}
        className={`popup__input popup__input_${fieldName}`}
        placeholder={placeholder}
        required
        minLength={minLength}
        maxLength={maxLength}
        value={value}
        onChange={onChange}
      />
      <span id={`${idName}-error`} className="popup__input_type_error">
        {(errorMessage as ErrorMessage)[name]}
      </span>
    </>
  );
};

export default Input;
