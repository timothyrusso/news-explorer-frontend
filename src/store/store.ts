import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
