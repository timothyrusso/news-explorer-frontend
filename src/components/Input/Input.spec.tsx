import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Input from './Input';

describe('Input component', () => {
  it('Should render the Input component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Input
          type="text"
          idName="testId"
          name="testName"
          fieldName="testField"
          placeholder="testPlaceholder"
          onChange={jest.fn()}
          minLength={10}
          maxLength={20}
          value="testValue"
          errorMessage={{ testName: 'Error message for testing' }}
          labelText="Test Label"
        />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
