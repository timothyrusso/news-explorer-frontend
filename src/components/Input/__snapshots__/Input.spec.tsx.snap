// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render the Input component 1`] = `
<DocumentFragment>
  <label
    class="popup__input-label"
    for="testId"
  >
    Test Label
  </label>
  <input
    class="popup__input popup__input_testField"
    id="testId"
    maxlength="20"
    minlength="10"
    name="testName"
    placeholder="testPlaceholder"
    required=""
    type="text"
    value="testValue"
  />
  <span
    class="popup__input_type_error"
    id="testId-error"
  >
    Error message for testing
  </span>
</DocumentFragment>
`;
